server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceShardsAllocatorTests.java

```diff
@@ -85,6 +86,21 @@ public void assertNoPendingReroute() {
         }
     }
 
+    private static class TestActionListener implements ActionListener<Void> {
+
+        private volatile boolean wasCalled = false;
+
+        @Override
+        public void onResponse(Void unused) {
+            wasCalled = true;
+        }
+
+        @Override
+        public void onFailure(Exception e) {
+
```

Comment:
Nit: can assert no failures here too:

```suggestion
            fail("should not be called");
```